### DISCUSSION TITLE
Generating .meta files after installation modified coq

### DISCUSSION THREAD

**Discussion by smirnoffalexx** [princeton-vl/CoqGym#62] [Q&A] [answered]
Hello!

I have tried to use coqgym with singularity and with built docker image (with the same versions), but I had the same problem as described in Issue (Location of meta files for dataset #7). I have installed modified coq from the repository, but after compilation my coq files, I didn't get any .meta files. Could you show the correct solution of the problem?

**Comment by yangky11** [comment, marked as answer]
You were probably using the wrong Coq installation for compiling those projects. Can you run `which coqc` right before `cd coq_projects && make && cd ..` to check?

**Reply by smirnoffalexx** [reply]
/root/.opam/4.07.1+flambda/bin/coqc

**Reply by yangky11** [reply]
This is not the modified Coq in this repo. Here we add the modified Coq to PATH
https://github.com/princeton-vl/CoqGym/blob/680a5df29d845045e08121b9b6989f57f2841d9d/install.sh#L19
, so that `which coqc` should show something like `...../CoqGym/coq/bin/coqc`. 

I'm not sure why it wasn't the case for you. But you could try manually adding the correct Coq to PATH before compiling the projects. Also, you may have to run `make clean` before compiling since you have partially compiled some projects using the wrong Coq.

**Reply by smirnoffalexx** [reply]
Thank you, it solved problem with your data (coq_projects), but with my own directory, it doesn’t work (maybe I have to write makefile properly). What I have to take into account while writing makefiles for my own code?

**Reply by yangky11** [reply]
I don't think you need to do anything special in makefiles. The only makefile I wrote is [coq_projects/makefile](https://github.com/princeton-vl/CoqGym/blob/master/coq_projects/makefile). I didn't change the makefiles of individual projects in [coq_projects](https://github.com/princeton-vl/CoqGym/tree/master/coq_projects).

To generate *.meta file, the only thing needed is to use the modified Coq. Because the *.meta files are generated here:
https://github.com/princeton-vl/CoqGym/blob/680a5df29d845045e08121b9b6989f57f2841d9d/coq/toplevel/vernac.ml#L213

**Reply by smirnoffalexx** [reply]
Thanks, make clean, and removing Makefile.coq.conf before make solved the problem.
